pkg/rancher-desktop/config/__tests__/settings.spec.ts

```diff
@@ -1,15 +1,193 @@
+          <string>sub-title</string>
+          <key>locked</key>
+          <true/>
+          <key>subvar</key>
+          <string>sub-var</string>
+        </dict>
+      </dict>
+    </dict>
+    <key>debug</key>
+    <true/>
+    <key>diagnostics</key>
+    <dict>
+      <key>showMuted</key>
+      <false/>
+      <key>locked</key>
+      <true/>
+      <key>mutedChecks</key>
+      <dict/>
+    </dict>
+  </dict>
+</plist>`;
+  const unlockedJSONProfile = JSON.stringify({
+    ignoreThis:      { soups: ['beautiful', 'vichyssoise'] },
+    containerEngine: { name: 'should be ignored' },
+    kubernetes:      { version: "Shouldn't see this" },
+  });
+  const lockedJSONProfile = JSON.stringify({
+    ignoreThis:      { soups: ['beautiful', 'vichyssoise'] },
+    containerEngine: {
+      imageAllowList: {
+        enabled:  true,
+        patterns: ['nginx', 'alpine'],
+      },
+    },
+    kubernetes: { version: "Shouldn't see this" },
+  });
+  const unlockedPlistProfile = `
+  <?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+  <dict>
+    <key>kubernetes</key>
+    <dict>
+      <key>version</key>
+      <string>Should be ignored</string>
+    </dict>
+  </dict>
+</plist>
+`;
+  const lockedPlistProfile = `<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+  <dict>
+    <key>version</key>
+    <integer>4</integer>
+    <key>containerEngine</key>
+    <dict>
+      <key>imageAllowList</key>
+      <dict>
+        <key>enabled</key>
+        <true/>
+        <key>not_schema</key>
+        <true/>
+        <key>patterns</key>
+        <array>
+        <string>nginx</string>
+        <string>alpine</string>
+        </array>
+      </dict>
+    </dict>
+    <key>kubernetes</key>
+    <dict>
+      <key>version</key>
+      <string>1.23.15</string>
+      <key>containerEngine</key>
+      <string>moby</string>
+      <key>enabled</key>
+      <true/>
+    </dict>
+    <key>testSettings</key>
+    <dict>
+      <key>testTitle</key>
+      <string>test-title</string>
+      <key>testStruct</key>
+      <dict>
+        <key>title</key>
+        <string>test-struct</string>
+        <key>subStruct</key>
+        <dict>
+          <key>title</key>
+          <string>sub-title</string>
+          <key>locked</key>
+          <true/>
+          <key>subvar</key>
+          <string>sub-var</string>
+        </dict>
+      </dict>
+    </dict>
+    <key>debug</key>
+    <true/>
+    <key>diagnostics</key>
+    <dict>
+      <key>showMuted</key>
+      <false/>
+      <key>locked</key>
+      <true/>
+      <key>mutedChecks</key>
+      <dict/>
+    </dict>
+  </dict>
+</plist>
+  `;
+  const lockedAccessors = ['containerEngine.imageAllowList.enabled', 'containerEngine.imageAllowList.patterns'];
+  let mock: any;
```

Comment:
Does it make sense to call `jest.afterEach(mock.restore())` at this level (so we don't have to do that every time)?

Reply:
Definitely. I had early problems with mocking so did it that way, but it can be done once.